cursor-cli-demo/components/terminal.tsx

```
@@ -0,0 +1,114 @@
+      </div>
+    );
+  }
+
+  const status = run?.status;
+  const isRunning = status === "EXECUTING";
+  const isQueued = status === "PENDING_VERSION" || status === "DELAYED" || !status;
+  const isFailed = status === "FAILED" || status === "CRASHED" || status === "SYSTEM_FAILURE";
+  const isComplete = status === "COMPLETED";
+
+  // Notify parent when run finishes
+  const notified = useRef(false);
```

Biome (2.3.13)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

Hooks should not be called after an early return.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)